### DISCUSSION TITLE
where is the StorageMgr be inited?

### DISCUSSION THREAD

**Discussion by EysaDun** [filecoin-project/lotus#6054] [General Q&A]
I am a chinese developer, and i'm sorry for my week English.
Here has a question block me:

func (sm *StorageMinerAPI) WorkerConnect(ctx context.Context, url string) error {
	w, err := connectRemoteWorker(ctx, sm, url)
	if err != nil {
		return xerrors.Errorf("connecting remote storage failed: %w", err)
	}

	log.Infof("Connected to a remote worker at %s", url)

	return sm.StorageMgr.AddWorker(ctx, w)
}

this code is Worker register to miner, i dont understand where StorageMgr init in line: sm.StorageMgr.AddWorker(ctx, w). 
WorkerConnect func is called by nodeApi, nodeApi has no StorageMgr.
which nice man could answer me, thank you very much!

**Comment by de-authority** [comment]
I saw this question is asked before. do you need further help. maybe you can ask in slack channel

**Comment by LexLuthr** [comment]
@EysaDun  Do you need further help with this?

**Reply by LexLuthr** [reply]
I am archiving this discussion due to no reply for more than 7 days.
